### ISSUE TITLE
Impossible to upload a new version of the experimental plugin

### ISSUE THREAD

**Issue by alexbruy** [qgis/QGIS-Django#477]
It is now impossible to upload a new version of the experimental plugin, the error is
```
There were errors reading plugin package (please check also your plugin's metadata).
The 'experimental' flag in the form does not match the 'experimental' flag in the plugins package metadata.
```

However, the upload form clearly states
```
Check this box if this version is experimental, leave unchecked if it's stable. Please note that this field might be overridden by metadata (if present).
```

The metadata file **HAS** the experimental metadata flag set, however it seems that existing logic was broken recently and override it not working anymore. Please advertise such changes as they are breaking workflows.

**Comment by Xpirix** [comment]
Hi @alexbruy,

Can you please provide the steps you took? I tested uploading experimental plugin versions three times and didn't encounter any errors: https://plugins.qgis.org/plugins/test_modul/#plugin-versions

Here are the steps I followed:

1. Uploaded a test plugin (first as experimental, then deleted it, and then uploaded it again as non-experimental).
2. Initialized a new version with experimental=True in the metadata.
3. Uploaded the new version with the experimental option checked in the form.
4. Initialized another version with experimental=True in the metadata.
5. Uploaded this last version with the experimental option checked in the form.

AFAIK no changes has been made recently that would break any workflow on the website.

**Comment by Xpirix** [comment]
I agree with you that the form description doesn't really match the expected workflow. I tested uploading a new experimental version without checking the Experimental checkbox and I got the error.

**Comment by alexbruy** [comment]
It is simple:

1. Have an experimental plugin
2. Prepare a new version of the plugin
3. Try to add new version by selecting plugin package and leaving everything else unchanged
4. Got an error as in the first comment.

This worked before without any errors, I never had any issues uploading new versions of the plugins (be it experimental or not) and all data were taken from the metadata file.

**Comment by Xpirix** [comment]
Thank you for the clarification. I will take a look into the issue further.

**Comment by Xpirix** [comment]
Hi @alexbruy , please find the proposed fix at #494
